### PR TITLE
Added Context for wrapping

### PR DESCRIPTION
**Pull request by Schaltfehler**
Hi @JohnSundell 
Context for wrapping as discussed [here](https://github.com/JohnSundell/Wrap/issues/9).
Please have a look and let me know what you think :-)

### CODE REVIEW

File: Sources/Wrap.swift [hunk 1]
```diff
@@ -170,15 +170,19 @@ public protocol WrappableEnum: WrapCustomizable {}
 public class Wrapper {
     private var dateFormatter: NSDateFormatter?
     
+    /// Any contextual object that was supplied when unboxing was started
```
Comment (JohnSundell) [inline Sources/Wrap.swift]: It should be "wrapping" here, instead of "unboxing" :)

File: Tests/WrapTests.swift [hunk 2]
```diff
@@ -564,12 +564,55 @@ class WrapTests: XCTestCase {
...
+                    return "age"
+                }
+                return propertyName
+            }
+        }
+        
+        do {
+            try VerifyDictionary(Wrap(Model()), againstDictionary: [
+                "int" : 27
+                ])
```
Comment (JohnSundell) [inline Tests/WrapTests.swift]: Nitpick, some incorrect indentation here :) Also further below on line `604`

File: Sources/Wrap.swift [hunk 3]
```diff
@@ -120,13 +120,13 @@ public protocol WrapCustomizable {
      *  Returning nil from this method will be treated as an error, and cause
      *  a `WrapError.WrappingFailedForObject()` error to be thrown.
      */
-    func wrap() -> AnyObject?
+    func wrap(context: Any?) -> AnyObject?
```
Comment (JohnSundell) [inline Sources/Wrap.swift:123]: I'm thinking if we want to use explicit parameter labeling here (Swift 3-style) to make the API more intuitive. Right now I have to know that the object I supply refers to `context`, but if we use a parameter label it'll be a lot more clear, like this:

```
// When used without the context parameter
wrap()
```

```
// When used with the context parameter
wrap(context: "A context")
```

What do you think @Schaltfehler?

**Comment by JohnSundell** [comment]
Hey @Schaltfehler, sorry for the late response! Added some comments inline, let me know what you think!

File: Sources/Wrap.swift (hunk 3 above)
Comment (Schaltfehler) [inline Sources/Wrap.swift:123]: Agree to that and was thinking about it too. 
But with Swift3 [all labels will show same behaviour](https://github.com/apple/swift-evolution/blob/master/proposals/0046-first-label.md), means the first parameter name will also be its external name and wouldn't need an explicit parameter name anymore.

We could add explicit parameter naming now to have Swift3 style naming under Swift2 and remove it later when swift3 is released.  But if we do so shouldn't it be consistent throughout the lib and should use swift3 naming for all functions? 
`Wrapper(_:context:)` -> `Wrapper(dateFormatter:context:)`,
`keyForWrappingPropertyNamed(_:context:)` -> `keyForWrapping(propertyName:context:)`,
...

Or should Swift3 Naming be introduced in one commit when swift3 is released and this lib supports it officially?

**Comment by Schaltfehler** [comment]
Thank you very much for your feedback. Applied minor fixes and commented and your swift3 style naming suggestion. :-)

**Comment by JohnSundell** [comment]
Would love to revive this on the `swift3` branch. @Schaltfehler you want to rebase against `swift3` (and use Swift 3) or should I?

**Comment by Schaltfehler** [comment]
The coming ~10days I am a bit busy so if you have to time, please go ahead. 
Otherwise I could do it after that :-)

**Comment by JohnSundell** [comment]
@Schaltfehler Included an implementation of this feature in Wrap 2.0: https://github.com/JohnSundell/Wrap/pull/17 :) Thanks a lot for your work on this!

**Comment by JohnSundell** [comment]
Implemented in Wrap 2.0 :)

**Pull request closed without merging**